lib/Doctrine/ORM/AbstractQuery.php

```diff
@@ -712,24 +654,10 @@ public function getQueryCacheProfile()
      * Change the default fetch mode of an association for this query.
      *
      * @param class-string $class
-     * @param string       $assocName
-     * @param int          $fetchMode
      * @psalm-param Mapping\ClassMetadata::FETCH_EAGER|Mapping\ClassMetadata::FETCH_LAZY $fetchMode
-     *
-     * @return $this
      */
-    public function setFetchMode($class, $assocName, $fetchMode)
+    public function setFetchMode(string $class, string $assocName, int $fetchMode): static
     {
-        if (! in_array($fetchMode, [Mapping\ClassMetadata::FETCH_EAGER, Mapping\ClassMetadata::FETCH_LAZY], true)) {
-            Deprecation::trigger(
-                'doctrine/orm',
-                'https://github.com/doctrine/orm/pull/9777',
-                'Calling %s() with something else than ClassMetadata::FETCH_EAGER or ClassMetadata::FETCH_LAZY is deprecated.',
-                __METHOD__
-            );
-            $fetchMode = Mapping\ClassMetadata::FETCH_LAZY;
-        }
-
```

Comment:
From https://github.com/doctrine/orm/pull/9777#discussion_r879878197 : 

> On 3.0, we would throw if the caller passed anything but the two allowed values, right?

Do we really want to do this? If yes, maybe we should require an assertion library :thinking: